### PR TITLE
Documentation for Kind

### PR DESCRIPTION
**Pull request by chemzqm**
It's useful for custom source.

### CODE REVIEW

File: doc/denite.txt [hunk 1]
```diff
@@ -1451,12 +1451,64 @@ abbr		(String)			(Optional)
 ==============================================================================
 CREATE KIND						*denite-create-kind*
 
-Note: Not implemented yet...
+Custom Kind can be created to contain extra actions for your custom source.
+Before create your new kind, you should have a look at existing kinds in
+rplugin/python3/denite/kind/*.py
+
+Instead of create new kind from sketch, you can extend one of the exiting
```
Comment (nelstrom) [inline doc/denite.txt]: To my ear, "from sketch" sounds weird. Did you mean "Instead of creating a new kind *from scratch"*?

Comment (e-kwsm) [inline doc/denite.txt]: s/exiting/existing/

Comment (chemzqm) [inline doc/denite.txt]: Yep, I've picked wrong word 😄

Comment (chemzqm) [inline doc/denite.txt]: thanks

File: doc/denite.txt [hunk 2]
```diff
@@ -1451,12 +1451,64 @@ abbr		(String)			(Optional)
...
+Before create your new kind, you should have a look at existing kinds in
+rplugin/python3/denite/kind/*.py
+
+Instead of create new kind from scratch, you can extend one of the existing
+Kind class, let's take `openable` for example: >
+
+	from ..kind.openable import Kind as Openable 
+
+	class Kind(Openable):
+		def __init__(self, vim):
```
Comment (lambdalisue) [inline doc/denite.txt]: It is an example code for Python. Why do you use 8 spaces (or tab?)

File: doc/denite.txt [hunk 3]
```diff
@@ -1451,12 +1451,64 @@ abbr		(String)			(Optional)
...
+			super().__init__(vim)
+			self.name = 'mykind'
+
+		def action_open(self, context):
+			target = context['targets'][0]
+			path = target['action__path']
+			" assume you have a MyEdit command
```
Comment (lambdalisue) [inline doc/denite.txt]: Wrong comment char. The line should starts from `#`

File: doc/denite.txt [hunk 4]
```diff
@@ -1451,12 +1451,64 @@ abbr		(String)			(Optional)
 ==============================================================================
 CREATE KIND						*denite-create-kind*
 
-Note: Not implemented yet...
+Custom Kind can be created to contain extra actions for your custom source.
+Before create your new kind, you should have a look at existing kinds in
+rplugin/python3/denite/kind/*.py
+
+Instead of create new kind from scratch, you can extend one of the existing
+Kind class, let's take `openable` for example: >
```
Comment (lambdalisue) [inline doc/denite.txt]: Vim's help does not use back-quote. I think it should be replaced with a double quote instead.

File: doc/denite.txt [hunk 5]
```diff
@@ -1451,12 +1451,64 @@ abbr		(String)			(Optional)
...
+					*denite-kind-attribute-redraw_actions*
+redraw_actions (List) 				(Optional)
+		List of action names which redraw Denite buffer after
+		executed.
+
+					*denite-kind-attribute-action_*
+action_{name} (Function) 			(Optional)
+		Custom action logic with {name} as action name. The function
+		is passed with {self} and {context}. To get the seleted
+		candidates, you could use: `context['targets']`
```
Comment (lambdalisue) [inline doc/denite.txt]: Same here

File: doc/denite.txt [hunk 6]
```diff
@@ -1451,12 +1451,64 @@ abbr		(String)			(Optional)
...
+			target = context['targets'][0]
+			path = target['action__path']
+			" assume you have a MyEdit command
+			self.vim.call('MyEdit', path)
+>
+After create new Kind class, alias it to your source by >
+
+	self.kind = Kind(vim)
+<
+in the {__init__} method of your source class.
```
Comment (lambdalisue) [inline doc/denite.txt]: Vim's help already use `{xxxxx}` syntax for an argument so it should not be used.

File: doc/denite.txt (hunk 2 above)
Comment (lambdalisue) [inline doc/denite.txt]: I noticed that an example code in `:h python` use a tab char..... I really don't like to use tab char for python that but if you prefer to use tab, I'm ok now.

Comment (chemzqm) [inline doc/denite.txt]: Got it, I will use copy&paste to get avoid of this problem next time.

**Comment by lambdalisue** [comment]
There are some left over for `{xxxxx}`

**Comment by chemzqm** [comment]
@lambdalisue  
```
		Custom action logic with {name} as action name. The function
		is passed with {self} and {context}. To get the seleted
```
the usage of `{xxxxx}` here is the same as the ones of `:h denite-filter-attribute-filter`

**Comment by lambdalisue** [comment]
Hum... OK.

**Comment by Shougo** [comment]
Nice work!
Merged.

**Pull request merged**